### PR TITLE
feat(MatchGenerate): split senders and receivers

### PR DESCRIPTION
**Pull request by AdrianAndersen**

### CODE REVIEW

**Comment by AdrianAndersen** [comment]
[Splitte MatcherSpec branches opp i senders og receivers](https://trello.com/c/P6XUZPnk/1065-splitte-matcherspec-branches-opp-i-senders-og-receivers)

File: src/collections/match/operations/match-generate-operation-helper.ts [hunk 1]
```diff
@@ -197,11 +198,15 @@ export function verifyMatcherSpec(
...
-    Array.isArray(m["userMatchLocations"]) &&
-    m["branches"].every(
+    Array.isArray(m["senders"]) &&
+    Array.isArray(m["receivers"]) &&
+    m["senders"].every(
+      (branchId) => typeof branchId === "string" && branchId.length === 24,
+    ) &&
+    m["receivers"].every(
       (branchId) => typeof branchId === "string" && branchId.length === 24,
     ) &&
```
Comment (LarsSelbekk) [inline src/collections/match/operations/match-generate-operation-helper.ts:208]: Use ObjectId-check

File: src/collections/match/operations/match-generate-operation-helper.ts [hunk 2]
```diff
@@ -20,7 +20,8 @@ import {
  * The information required to generate matches.
  */
 export interface MatcherSpec {
-  branches: string[];
+  senders: string[];
+  receivers: string[];
```
Comment (LarsSelbekk) [inline src/collections/match/operations/match-generate-operation-helper.ts:24]: I prefer senderBranches and receiverBranches like we had before, as this makes me expect user IDs.

**Review by LarsSelbekk — CHANGES_REQUESTED** [review]
Nits

**Review by LarsSelbekk — APPROVED** [review]

**Pull request merged**